### PR TITLE
support client encryption

### PR DESCRIPTION
**Pull request by amoskong**
Signed-off-by: Amos Kong <amos@scylladb.com>

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -1402,6 +1402,9 @@ def config_setup(self, seed_address=None, cluster_name=None, enable_exp=True, en
             self.remoter.send_files(src='./data_dir/ssl_conf',
                                     dst='/tmp/')
             self.remoter.run('sudo mv /tmp/ssl_conf/*.* /etc/scylla/')
+            self.remoter.run('sudo cp -r /tmp/ssl_conf/unittest /etc/scylla/')
+            self.remoter.run('mkdir -p ~/.cassandra/')
```
[the remark this answers is not in the document]
Comment (ShlomiBalalis) [reply sdcm/cluster.py]: @roydahan The -p flag already avoids such errors

File: sdcm/cluster.py [hunk 2]
```diff
@@ -1402,6 +1402,9 @@ def config_setup(self, seed_address=None, cluster_name=None, enable_exp=True, en
             self.remoter.send_files(src='./data_dir/ssl_conf',
                                     dst='/tmp/')
             self.remoter.run('sudo mv /tmp/ssl_conf/*.* /etc/scylla/')
+            self.remoter.run('sudo cp -r /tmp/ssl_conf/unittest /etc/scylla/')
```
[the remark this answers is not in the document]
Comment (amoskong) [reply sdcm/cluster.py]: The `*.*` pattern can't match `unittest/` directory.
There are some unused directory insider ssl_conf/, so I didn't use `mv /tmp/ssl_conf/* /etc/scylla/`.

File: sdcm/cluster.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (amoskong) [reply sdcm/cluster.py]: > @roydahan The -p flag already handles such errors

Yes. Thanks

File: sdcm/cluster.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (roydahan) [reply sdcm/cluster.py]: If they are unused why do we have them?
And what not just move all?
It reduce the copy/move to one command.

[the remark this answers is not in the document]
Comment (amoskong) [reply sdcm/cluster.py]: We saved all files of generating encryption key in to `data_dir/ssl_conf/example/`, which will be used for updating the key. Only the generated key is copied to `data_dir/ssl_conf/`, which will be copied to db node in testing.

Current:
```
$ ls data_dir/ssl_conf/example/
cadb.key  cadb.pem  cadb.srl  db.cfg  db.crt  db.csr  db.key
```

How about move data_dir/ssl_conf/example/ to data_dir/ssl_conf_history/? then we can use one move to copy all configure files to /etc/scylla/?

**Comment by amoskong** [comment]
The latest PR had been tested by --- Test ID: 8263c912-4d58-4683-8ac0-9b4de1b44a60   
image

File: test-cases/longevity/longevity-1TB-7days-authorization-and-tls-ssl-1dis-2nondis-nemesis.yaml [hunk 3]
```diff
@@ -32,8 +32,7 @@ user_prefix: 'longevity-tls-1tb-7d-1dis-2nondis'
 failure_post_behavior: destroy
 space_node_threshold: 644245094
 server_encrypt: 'true'
-# Setting client encryption to false for now, till we will find the way to make c-s work with that
-client_encrypt: 'false'
+client_encrypt: True
```
Comment (bentsi) [inline test-cases/longevity/longevity-1TB-7days-authorization-and-tls-ssl-1dis-2nondis-nemesis.yaml]: for consistency let's use: `client_encrypt: true` instead "True" (with uppercase)
also no need to put single quotation marks around the "true" 
should be `server_encrypt: true`

File: sdcm/cluster.py [hunk 4]
```diff
@@ -1408,23 +1408,26 @@ def config_setup(self, seed_address=None, cluster_name=None, enable_exp=True, en
         if server_encrypt or client_encrypt:
             self.remoter.send_files(src='./data_dir/ssl_conf',
                                     dst='/tmp/')
-            self.remoter.run('sudo mv /tmp/ssl_conf/*.* /etc/scylla/')
+            self.remoter.run('mkdir -p ~/.cassandra/')
```
Comment (bentsi) [inline sdcm/cluster.py]: same code is used in `node_setup`. Why?

File: sdcm/cluster.py [hunk 5]
```diff
@@ -1408,23 +1408,26 @@ def config_setup(self, seed_address=None, cluster_name=None, enable_exp=True, en
...
+            self.remoter.run('cp /tmp/ssl_conf/unittest/cqlshrc ~/.cassandra/')
```
Comment (bentsi) [inline sdcm/cluster.py]: why the `unittest` directory is in the path??

Comment (amoskong) [reply sdcm/cluster.py]: There are some key files that generated from scylla unittest, the keys are used in client side.
Maybe just rename it to `ssl_conf/client/..`

File: test-cases/longevity/longevity-1TB-7days-authorization-and-tls-ssl-1dis-2nondis-nemesis.yaml (hunk 3 above)
Comment (amoskong) [reply test-cases/longevity/longevity-1TB-7days-authorization-and-tls-ssl-1dis-2nondis-nemesis.yaml]: Fixed.

File: sdcm/cluster.py (hunk 4 above)
Comment (amoskong) [reply sdcm/cluster.py]: Thats' the node_setup of loader.
We run c-s on loader,  run cqlsh & c-s on db node.

File: sdcm/cluster.py [hunk 6]
```diff
@@ -2917,6 +2921,13 @@ def node_setup(self, node, verbose=False, db_node_address=None, **kwargs):
         collectd_setup = ScyllaCollectdSetup()
         collectd_setup.install(node)
         self.install_gemini(node=node)
+        node.remoter.send_files(src='./data_dir/ssl_conf',
+                                dst='/tmp/')
+        node.remoter.run('mkdir -p ~/.cassandra/')
```
[the remark this answers is not in the document]
Comment (amoskong) [reply sdcm/cluster.py]: Fixed.

File: sdcm/cluster.py [hunk 7]
```diff
@@ -2917,6 +2921,13 @@ def node_setup(self, node, verbose=False, db_node_address=None, **kwargs):
         collectd_setup = ScyllaCollectdSetup()
         collectd_setup.install(node)
         self.install_gemini(node=node)
+        node.remoter.send_files(src='./data_dir/ssl_conf',
```
[the remark this answers is not in the document]
Comment (amoskong) [reply sdcm/cluster.py]: Only for client_encrypt is enabled.

**Comment by roydahan** [comment]
@amoskong please review and merge

**Comment by amoskong** [comment]
> @amoskong please review and merge

/CC @bentsi

**Comment by roydahan** [comment]
@bentsi please re-review and merge if you approve

**Comment by amoskong** [comment]
@bentsi ping

```
<Quote from hangout>
There is a pressure around the client encryption as part of the longevity.
We suspect there was a problem at a customer system with regarding to that.
If you can give this high priority and make it work for longevity and send PR to enable it for longevities 1TB and 200GB.
```

**Pull request merged**